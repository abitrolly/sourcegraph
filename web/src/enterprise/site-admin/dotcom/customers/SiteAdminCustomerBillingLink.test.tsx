import React from 'react'
import { SiteAdminCustomerBillingLink } from './SiteAdminCustomerBillingLink'
import { mount } from 'enzyme'

jest.mock('mdi-react/ExternalLinkIcon', () => 'ExternalLinkIcon')

describe('SiteAdminCustomerBillingLink', () => {
    test('linked billing account', () => {
        expect(
            mount(
                <SiteAdminCustomerBillingLink
                    customer={{ id: 'u', urlForSiteAdminBilling: 'https://example.com' }}
                    onDidUpdate={() => undefined}
                />
            )
        ).toMatchSnapshot()
    })

    test('no linked billing account', () => {
        expect(
            mount(
                <SiteAdminCustomerBillingLink
                    customer={{ id: 'u', urlForSiteAdminBilling: null }}
                    onDidUpdate={() => undefined}
                />
            )
        ).toMatchSnapshot()
    })
})
