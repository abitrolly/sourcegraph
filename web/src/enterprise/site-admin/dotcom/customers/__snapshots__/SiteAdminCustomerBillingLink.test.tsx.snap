// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteAdminCustomerBillingLink linked billing account 1`] = `
<SiteAdminCustomerBillingLink
  customer={
    Object {
      "id": "u",
      "urlForSiteAdminBilling": "https://example.com",
    }
  }
  onDidUpdate={[Function]}
>
  <div
    className="site-admin-customer-billing-link"
  >
    <div
      className="d-flex align-items-center"
    >
      <a
        className="mr-2 d-flex align-items-center"
        href="https://example.com"
      >
        View customer account 
        <ExternalLinkIcon
          className="icon-inline ml-1"
        />
      </a>
      <button
        className="btn btn-secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Unlink
      </button>
    </div>
  </div>
</SiteAdminCustomerBillingLink>
`;

exports[`SiteAdminCustomerBillingLink no linked billing account 1`] = `
<SiteAdminCustomerBillingLink
  customer={
    Object {
      "id": "u",
      "urlForSiteAdminBilling": null,
    }
  }
  onDidUpdate={[Function]}
>
  <div
    className="site-admin-customer-billing-link"
  >
    <div
      className="d-flex align-items-center"
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Link billing customer
      </button>
    </div>
  </div>
</SiteAdminCustomerBillingLink>
`;
